tap_google_ads/streams.py

```diff
@@ -80,9 +80,60 @@ def build_parameters():
     param_str = ",".join(f"{k}={v}" for k, v in API_PARAMETERS.items())
     return f"PARAMETERS {param_str}"
 
+def generate_where_and_orderby_clause(last_pk_fetched, filter_param, composite_pks):
+    """
+    Generates a WHERE clause and a ORDER BY clause based on filter parameter(`key_properties`), and
+    `last_pk_fetched`.
+
+    Example:
+
+    Single PK Case:
+
+    filter_param = 'id'
+    last_pk_fetched = 1
+    composite_pks = False
+    Returns:
+    WHERE id > 1 ORDER BY id ASC
+
+    Composite PK Case:
+
+    composite_pks = True
+    filter_param = 'id'
+    last_pk_fetched = 1
+    Returns:
+    WHERE id >= 1 ORDER BY id ASC
+    """
+    where_clause = ""
+    order_by_clause = ""
+
+    # Even If the stream has a composite primary key, we are storing only a single pk value in the bookmark.
+    # So, there might be possible that records with the same single pk value exist with different pk value combinations.
+    # That's why for composite_pks we are using a greater than or equal operator.
+    comparison_operator = ">="
```

Comment:
I think there was a conversation around this, but I just want to make sure we don't want to store both pieces of the composite key in the bookmark. I think there might be a couple of hard-to-reach edge cases with this approach, but they definitely don't seem likely.

Reply:
@atribed, I Agree with you. If the stream has a composite primary key, then we are storing only a single pk value in the bookmark. There might be a duplication of records for streams that have composite pks.